flow360/component/simulation/time_stepping/time_stepping.py

```diff
@@ -0,0 +1,139 @@
+
+    @classmethod
+    def default_steady(cls):
+        """
+        returns default steady Ramp CFL settings
+        """
+        return cls(initial=5, final=200, ramp_steps=40)
+
+
+class AdaptiveCFL(Flow360BaseModel):
+    """
+    Adaptive CFL for time stepping component
+    """
+
+    type: Literal["adaptive"] = pd.Field("adaptive", frozen=True)
+    min: Optional[pd.PositiveFloat] = pd.Field(default=0.1)
+    max: Optional[pd.PositiveFloat] = pd.Field(None)
+    max_relative_change: Optional[pd.PositiveFloat] = pd.Field(None)
+    convergence_limiting_factor: Optional[pd.PositiveFloat] = pd.Field(None)
+
+    @classmethod
+    def default_unsteady(cls):
+        """
+        returns default unsteady Adaptive CFL settings
+        """
+        return cls(max=1e6, convergence_limiting_factor=1.0, max_relative_change=50)
+
+    @classmethod
+    def default_steady(cls):
+        """
+        returns default steady Adaptive CFL settings
+        """
+        return cls(max=1e4, convergence_limiting_factor=0.25, max_relative_change=1)
+
+
+class BaseTimeStepping(Flow360BaseModel, metaclass=ABCMeta):
+    """
+    Base class for time stepping component
+    """
+
+    max_pseudo_steps: Optional[int] = pd.Field(2000, gt=0, le=100000)
+    order_of_accuracy: Optional[Literal[1, 2]] = pd.Field(2)
+    CFL: Optional[Union[RampCFL, AdaptiveCFL]] = pd.Field(None)
+
+    #  TODO:
+    # # pylint: disable=arguments-differ
+    # def to_solver(self, params, **kwargs) -> BaseTimeStepping:
+    #     """
+    #     returns configuration object in flow360 units system
+    #     """
+    #     return super().to_solver(params, **kwargs)
+
+
+class SteadyTimeStepping(Flow360BaseModel):
+    """
+    Steady time stepping component
+    """
+
+    model_type: Literal["Steady"] = pd.Field("Steady", frozen=True)
+    physical_steps: Literal[1] = pd.Field(1, frozen=True)
+    time_step_size: Literal["inf"] = pd.Field("inf", frozen=True)
+    CFL: Optional[Union[RampCFL, AdaptiveCFL]] = pd.Field(
+        default=AdaptiveCFL.default_steady(),
+    )
+
+    @pd.model_validator(mode="before")
+    @classmethod
+    def set_default_cfl(cls, values):
+        """
+        Populate CFL's None fields with default
+        """
+        if "CFL" not in values:
+            return values  # will be handeled by default value
+        cfl_input = values["CFL"]
+        if isinstance(cfl_input, AdaptiveCFL):
+            cfl_input = _apply_default_to_none(cfl_input, AdaptiveCFL.default_steady())
+        elif isinstance(cfl_input, RampCFL):
+            cfl_input = _apply_default_to_none(cfl_input, RampCFL.default_steady())
+        return values
+
+
+class UnsteadyTimeStepping(Flow360BaseModel):
+    """
+    Unsteady time stepping component
+    """
+
+    model_type: Literal["Unsteady"] = pd.Field("Unsteady", frozen=True)
+    physical_steps: pd.PositiveInt = pd.Field()
+    time_step_size: TimeType.Positive = pd.Field()
```

Comment:
I think these 2 parameter names can be unified:
```
time_steps
time_step_size
```
or
```
steps
step_size
```

Reply:
again it is collision with max_psuedo_step.

Reply:
My proposition was to have different names for this parameter on steady and unsteady. If you think of using only one of these classes, parameter names are very confusing. They only have sense when you look at both of the classes at the same time.
```python
UnsteadyTimeStepping(physical_steps=10,  time_step_size=1, max_pseudo_steps=10)
```
These are lengthy parameter names. Instead I would make parameters that are more often used or required, to be easier available. For example:

```python
UnsteadyTimeStepping(steps=10, step_size=1, max_pseudo_steps=10)
```

Similarly, steady time stepping does not need to be considered in the context of unsteady:
```python
SteadyTimeStepping(max_pseudo_steps=10)
```
instead, I think better it is:
```python
SteadyTimeStepping(max_steps=10)
```

There is no single parameter that has two meanings in two contexts so I think there is no collision. It is just a proposition, if you think this will be very confusing we don't need to do it. Just don't think about current users, think about new users.

Reply:
we can do this.

Reply:
yes, I like the `Steady`